psdk_wrapper/include/psdk_wrapper/json_utils.hpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2024 Unmanned Life
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/**
+ * @file json_utils.hpp
+ *
+ * @brief Header file containing utility functions for C++ library nlohmann_json
+ *
+ * @authors Alejandro Mora
+ * Contact: alejandro@unmanned.life
+ *
+ */
+#ifndef PSDK_WRAPPER_INCLUDE_PSDK_WRAPPER_JSON_UTILS_HPP_
+#define PSDK_WRAPPER_INCLUDE_PSDK_WRAPPER_JSON_UTILS_HPP_
+
+#include <dji_hms.h>
+
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+#include "psdk_interfaces/msg/hms_info_msg.hpp"
+#include "psdk_interfaces/msg/hms_info_table.hpp"
+
+namespace psdk_ros2
+{
+namespace json_utils
+{
+inline nlohmann::json
+parse_file(const std::string& path)
+{
+  std::ifstream file(path);
+  return nlohmann::json::parse(file);
+}
+
+template <typename T>
+inline std::string
+to_hex_str(const T& value, const bool& is_lower_case = true)
+{
+  std::stringstream stream;
+  stream << "0x" << std::setfill('0') << std::setw(sizeof(T) * 2) << std::hex
+         << (is_lower_case ? std::nouppercase : std::uppercase) << value;
+  return stream.str();
+}
+
+inline psdk_interfaces::msg::HmsInfoTable
```

Comment:
As described [here](https://www.geeksforgeeks.org/inline-functions-cpp/), the compiler will ignore your request of `inline` since you have a for loop. Normally as a rule of thumb it doesn't make sense to make a function inline if its considerably long, because it creates more overhead than simply calling a regular function